file.go

```diff
@@ -57,14 +55,11 @@ type File struct {
 	treatasclean bool // Window Clean tests should succeed if set. [private]
 
 	// Observer pattern: many Text instances can share a File.
-	curtext *Text
-	text    []*Text // [private I think]
+	*observableeditablebuffer //make this a pointer
```

Comment:
Composition over embedding please. I am increasingly convinced that the right path (i.e. will end up with the best outcome and least work on your part) will be to have `ObservableEditableBuffer` wrap a `File` as an internal implementation detail. Remember that the high-level goal is to end up with a `File` that can be deleted and replaced with an `undo.Buffer`.